test/tests/Platform/WebLoginClient.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { IClientResponse, AccessTokenResponse } from '../../../model';
+import { HttpClient, IRequestParams } from '../../../src/HttpClient';
+import { WebLoginClient } from '../../../src/Platform/WebLoginClient';
+
+describe('WebLoginClient', () => {
+    let webLoginClient: WebLoginClient;
+    let httpClient: HttpClient;
+    const sessionId: string = '1234567890';
+
+    beforeEach(() => {
+        httpClient = {
+            doRequest: jest.fn(),
+            pollURLForTime: jest.fn(),
+        } as unknown as HttpClient;
+        webLoginClient = new WebLoginClient(httpClient, undefined);
```

Comment:
I'd prefer using "nock" to mock REST APIs and not Jest's mocks. Nock allows us to test end-to-end, without mocking functions like you did in this file.

```suggestion
        platformClient = new PlatformClient(PLATFORM_URL, undefined);
```